### PR TITLE
[*] Onboarding - Catch correctly authentication error on mongodb

### PR DESCRIPTION
**Pull request by VincentMolinie**

### CODE REVIEW

File: services/database-analyzer.js [hunk 1]
```diff
@@ -204,27 +204,41 @@ function DatabaseAnalyzer(databaseConnection, config, allowWarning) {
...
+          logger.error(
+            'Your database is empty.',
+            'Please, create some collections before running generate command.',
+          );
+          await eventSender.notifyError('database_empty', 'Your database is empty.', {
+            orm: 'mongoose',
+            dialect: 'mongodb',
+          });
+          return process.exit(1);
+        }
```
Comment (VincentMolinie) [inline services/database-analyzer.js]: I just added these lines in this file, the rest is unchanged ;)

Comment (arnaudbesnier) [reply services/database-analyzer.js]: @VincentMolinie @larcin if we consider the DRY principle here, this code cannot be accepted.
The error handling logic is shared with SQL database "onboarding" so there should not be code duplication.

@VincentMolinie we cannot basic copy paste in the code base anymore. You **must** consider this for any contributions.
@larcin this is smelly code, as this kind of logic was already handled for SQL databases.
A basic search for "Your database is empty" would have shown the code duplication.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Change Log
 
 ## [Unreleased]
+### Fixed
+- Onboarding - Catch correctly authentication error on mongodb.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]:
```suggestion
- Generate Command - Catch correctly authentication error on mongodb.
```
I would say "Installation" or "Generate Command" in the Lumber context.

File: services/database.js [hunk 3]
```diff
@@ -56,7 +56,8 @@ function Database() {
       }
 
       return MongoClient.connect(connectionUrl, connectionOptionsMongoClient)
-        .then(client => client.db(options.dbName));
+        .then(client => client.db(options.dbName))
+        .catch(error => handleAuthenticationError(error));
```
Comment (arnaudbesnier) [inline services/database.js]: This is a greatly isolated logic 👍

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,10 @@
 
 ### Fixed
 - MSSQL - Prevent potential bad database connection using upgrading an old version of `tedious` dependency.
+- Onboarding - Catch correctly authentication error on mongodb.
+- Onboarding - Catch correctly empty collections on mongodb.
+- Command Generate - Catch correctly authentication error on mongodb.
+- Command Generate - Catch correctly empty collections on mongodb.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: Why do you have 4 lines of changes?
No auto-review?

Comment (VincentMolinie) [reply CHANGELOG.md]: I did review my self But I did not had these 4 lines 🤔

**Pull request merged**